build/patch-charts.sh

```diff
@@ -333,10 +333,10 @@ function patchSidecarInjector() {
         - SETUID
   }' ${HELM_DIR}/istio-control/istio-discovery/files/injection-template.yaml
 
-  sed_wrap -i -e 's/runAsUser: 1337/runAsUser: {{ .ProxyUID }}/g' ${HELM_DIR}/istio-control/istio-discovery/files/injection-template.yaml
-  sed_wrap -i -e 's/runAsGroup: 1337/runAsGroup: {{ .ProxyUID }}/g' ${HELM_DIR}/istio-control/istio-discovery/files/injection-template.yaml
-  sed_wrap -i -e 's/fsGroup: 1337/fsGroup: {{ .ProxyGID }}/g' ${HELM_DIR}/istio-control/istio-discovery/files/injection-template.yaml
-  sed_wrap -i -e 's/fsGroup: 1337/fsGroup: {{ .ProxyGID }}/g' ${HELM_DIR}/istio-control/istio-discovery/files/gateway-injection-template.yaml
+  sed_wrap -i -E -e 's/(^ *)(runAsUser: )1337/\1{{ if .ProxyUID }}\n\1\2{{ .ProxyUID }}\n\1{{ end }}/g' \
+    -e 's/(^ *)(runAsGroup: )1337/\1{{ if .ProxyUID }}\n\1\2{{ .ProxyUID }}\n\1{{ end }}/g' \
+    -e 's/(^ *)(fsGroup: )1337/\1{{ if .ProxyGID }}\n\1\2{{ .ProxyGID }}\n\1{{ end }}/g' "${HELM_DIR}/istio-control/istio-discovery/files/injection-template.yaml"
```

Comment:
For consistency, let's ensure that the unnecessary whitespace is removed, just like it is in all other places in the injection-template.
```suggestion
  sed_wrap -i -E -e 's/(^ *)(runAsUser: )1337/\1{{- if .ProxyUID }}\n\1\2{{ .ProxyUID }}\n\1{{- end }}/g' \
    -e 's/(^ *)(runAsGroup: )1337/\1{{- if .ProxyUID }}\n\1\2{{ .ProxyUID }}\n\1{{- end }}/g' \
    -e 's/(^ *)(fsGroup: )1337/\1{{- if .ProxyGID }}\n\1\2{{ .ProxyGID }}\n\1{{- end }}/g' "${HELM_DIR}/istio-control/istio-discovery/files/injection-template.yaml"
```